ckanext/data_qld/commands.py

```diff
@@ -220,10 +220,22 @@ def _update_resource(res_dict):
                 print('Resource exception: %s' % ex)
 
         def _check_for_null_values(res, pkg_dict):
-            print('- Setting nature_of_change for resource %s' % res['id'])
-            res['nature_of_change'] = 'edit-resource-with-no-new-data'
-            if res.get('resource_visibility', None):
-                res['resource_visibility'] = 'Appropriate steps have been taken to minimise personal information re-identification risk prior to publishing'
+            if res.get('nature_of_change', None) is None:
+                print('- Setting nature_of_change for resource %s' % res['id'])
+                res['nature_of_change'] = 'edit-resource-with-no-new-data'
+            if res.get('update_frequency', None):
```

Comment:
per https://github.com/salsadigitalauorg/ckanext-data-qld/blob/aebf4f9b1d28fb74dca51f21bfd0db65b58437e4/ckanext/data_qld/commands.py#L189

please add counts for each of the types of updates you are doing and output them at the end on line 251

print on every 10th item so we don't have terminal buffer sowdowns, this will dramatically speed up operations, there is no need to print every update as it will slow down the operation.  i.e. https://realpython.com/python-modulo-operator/#modulo-operator-with-int when modulo on 10 = 0 then print with number.

at end print totals.

Reply:
There is still some issues with few scenarios and im keeping those messages just for development logs. I will remove once we got all sorted